shared/js/media/jpeg_metadata_parser.js

```diff
@@ -117,7 +121,8 @@ function parseJPEGMetadata(file, metadataCallback, metadataError) {
       var exif = parseEXIFData(data);
 
       if (exif.THUMBNAIL && exif.THUMBNAILLENGTH) {
-        var start = data.sliceOffset + data.viewOffset + 10 + exif.THUMBNAIL;
+        var start = data.sliceOffset + data.viewOffset + startOffset +
```

Comment:
From a quick reading, it is easy to think that startOffset is still 0 here and has not been set to 10.  The code is correct, but confusing.  I'd prefer it if you just set startOffset to 10 up above and remove the increment call below.